tests/utils/test_sentinel.py

```diff
@@ -0,0 +1,48 @@
+from copy import copy, deepcopy
+from pickle import loads, dumps
+from unittest import TestCase
+from weakref import ref
+
+from zipline.utils.sentinel import sentinel
+
+
+class SentinelTestCase(TestCase):
+    def tearDown(self):
+        sentinel._cache.clear()  # don't pollute cache.
+
+    def test_name(self):
+        self.assertEqual(sentinel('a').__name__, 'a')
+
+    def test_doc(self):
+        self.assertEqual(sentinel('a', 'b').__doc__, 'b')
+
+    def test_doc_differentiates(self):
```

Comment:
I'm not sure this is useful behavior.  I'd argue that this should just be an error.